src/test/java/seedu/address/storage/JsonAdaptedModuleClassTest.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.storage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.address.storage.JsonAdaptedModuleClass.MISSING_FIELD_MESSAGE_FORMAT;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalModuleClass.CS2103T_TUTORIAL;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.components.Name;
+
+public class JsonAdaptedModuleClassTest {
+
+    private static final String INVALID_NAME = "CS3230@Tutorial";
+
+    private static final String VALID_NAME = CS2103T_TUTORIAL.getName().toString();
+    private static final List<JsonAdaptedUuid> VALID_STUDENT_IDS = CS2103T_TUTORIAL.getStudentIds().stream()
+            .map(JsonAdaptedUuid::new)
+            .collect(Collectors.toList());
+
```

Comment:
Should you insert a `INVALID_STUDENT_IDS` list (and test case) also? Perhaps you can use a string that has less than 4 hyphens

Reply:
Currently since we are using UUID.randomUUID(), the edge case that could potentially happen is a null UUID rather than that of a UUID that is not specified properly. Hence i decided to only test for toModelType_nullStudentIds_throwsIllegalValueException()

Any thoughts on this?

Reply:
Resolved by introducing checks for invalid UUIDs